extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -574,18 +655,8 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 	}
 
 	getParent(element?: BookTreeItem): vscode.ProviderResult<BookTreeItem> {
-		if (element?.uri) {
-			let parentPath: string;
-			let contentFolder = element.book.version === BookVersion.v1 ? path.join(element.book.root, content) : element.book.root;
-			parentPath = path.join(contentFolder, element.uri.substring(0, element.uri.lastIndexOf(path.posix.sep)));
-			if (parentPath === element.root) {
-				return undefined;
-			}
-			let parentPaths = Array.from(this.currentBook.getAllNotebooks()?.keys()).filter(x => x.indexOf(parentPath) > -1);
-			return parentPaths.length > 0 ? this.currentBook.getAllNotebooks().get(parentPaths[0]) : undefined;
-		} else {
-			return undefined;
-		}
+		// Remove it for perf issues.
```

Comment:
In which cases does this get called? Does this affect the notebook being selected in the viewlet while navigating to a different notebook?

Reply:
The method getParent is not used by us in the BookTreeView. The vscode api requires us to implement it for the reveal method, but it never uses it.

Reply:
How is this a perf issue then if it never gets called?

If we don't need reveal then yeah I think it should be fine to remove. But you should update the return type if so (and remove the comment - it doesn't really help anything without additional context)

Reply:
getParent is optional on the interface but when the reveal method in extHostTreeView.ts is called, it throws an error if it's not implemented.